### ISSUE TITLE
circular-progress and circular-progress-four-color

### ISSUE THREAD

**Issue by srghma** [aforemny/material-components-web-elm#119]
HOW TO IMPLEMENT

https://github.com/material-components/material-components-web/tree/master/packages/mdc-circular-progress

https://material-components.github.io/material-components-web-components/demos/circular-progress/

https://github.com/material-components/material-components-web-components/blob/master/packages/circular-progress/README.md

https://github.com/material-components/material-components-web-components/blob/master/packages/circular-progress-four-color/README.md

______

@aforemny 

could you help with how the authoring should be done

for example I see you are duplicating [components](https://github.com/aforemny/material-components-web-elm/blob/master/src/chip/component.ts) and [foundations](https://github.com/aforemny/material-components-web-elm/blob/master/src/chip/foundation.ts) from https://github.com/material-components/material-components-web/blob/master/packages/mdc-chips/chip/component.ts and https://github.com/material-components/material-components-web/blob/master/packages/mdc-chips/chip/foundation.ts

![2020-08-25-11:03:13-screenshot](https://user-images.githubusercontent.com/7573215/91148626-98cf7c80-e6c2-11ea-95a2-85348c4cf90d.png)

1. is anything changed OR these are old files from material-components/material-components-web (I didnt search in history https://github.com/material-components/material-components-web) ?

_________

IF these files are a copy of material-components/material-components-web file WITH changes related to material-components-web-elm

THEN

maybe it's possible to derive these files from .patch files to keep them in sync with material-components/material-components-web original files AND see what was changed

e.g. `component.ts in this lib = component.ts file from material-components-web + component.ts.patch in this lib`

currently in chip dir

```
component.ts
custom-element.js
foundation.ts
```

expected:
```
component.ts.patch
custom-element.js
foundation.ts.patch
```

and in `scripts/update_files_from_material-components-web.sh`

```
set -e

update_file () {
  curl $https://github.com/material-components/material-components-web/blob/master/$1 -o ./$2
  patch ./$2 ./$2.patch
}

update_file packages/mdc-chips/chip/component.ts src/chip/component.ts
update_file packages/mdc-chips/chip/foundation.ts src/chip/foundation.ts
```

**Comment by srghma** [comment]
@aforemny friendly ping

**Comment by aforemny** [comment]
Hi @srghma, the .ts files that we have in our repository are very minimal patches to the respective upstream files. See the scripts `./bin/make-patches.sh` and `./bin/apply-patches.sh` that I use currently to update MDC Web.

This is a bit of a temporary work-around because I was not sure whether I want to patch component.ts and/ or foundation.ts. I have now come to the conclusion that I do want to patch both files slightly. So in the long term, I will probably fork MDC Web and maintain my patches there.

With regards to the circular progress component, I would not expect any necessity to patch anything (I have not yet looked at the source yet though.) Possibly, we would want the copy the files and adapt the imports for consistency only.

I am planning to update this library to MDC Web 5.0.0 within the next month, so I will have to add the circular progress indicator shortly.

When comparing patches (make-patches.sh), be sure that the material-components-web submodule is checked out which points to the release version of MDC Web that we are using.

I hope I have not missed anything in your question. If so, let me know. :)

**Comment by srghma** [comment]
> So in the long term, I will probably fork MDC Web and maintain my patches there

cool, would like to see

> I am planning to update this library to MDC Web 5.0.0

good, it will be easy to update to 7.0 then

**Comment by aforemny** [comment]
Just a heads-up: this is currently underway in f095005deb3a7263890903be700ad1bf27f591ed.

**Comment by aforemny** [comment]
Implemented in 19717c2189dff1e33ae03ae7a97cea79d5f262de.
